psycopg/psycopg/connection_anyio.py

```diff
@@ -0,0 +1,81 @@
+"""
+psycopg async connection objects using AnyIO
+"""
+
+# Copyright (C) 2022 The Psycopg Team
+
+
+from typing import Any, Optional, TYPE_CHECKING
+
+from . import errors as e
+from . import waiting_anyio
+from .abc import PQGen, PQGenConn, RV
+from .connection_async import AsyncConnection
+from .copy_anyio import AnyIOLibpqWriter
+from .rows import Row
+
+if TYPE_CHECKING:
+    import anyio
+    import sniffio
+else:
+    anyio = None
+    sniffio = None
+
+
+def _import_anyio() -> None:
```

Comment:
I propose an `@lru_cache()` here to avoid triggering the import machinery at every connection.

Reply:
If the module is in `sys.modules` it's already as fast as a dictionary lookup

Reply:
The function is not slow, but with a cache it's 4x faster for free.

```
In [4]: %timeit _import_anyio()
229 ns ± 4.45 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [5]: from functools import lru_cache

In [6]: _import_anyio2 = lru_cache()(_import_anyio)

In [7]: %timeit _import_anyio2()
52.4 ns ± 0.685 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```